src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -68,22 +70,11 @@ public class WebScriptHostManager : ScriptHostManager
             _hostRunningPollIntervalMilliseconds = hostPollingIntervalMilliseconds;
             _router = router;
 
-            var systemEventGenerator = config.HostConfig.GetService<IEventGenerator>() ?? new EventGenerator();
-            var systemTraceWriter = new SystemTraceWriter(systemEventGenerator, settingsManager, TraceLevel.Verbose);
-            if (config.TraceWriter != null)
-            {
-                config.TraceWriter = new CompositeTraceWriter(new TraceWriter[] { config.TraceWriter, systemTraceWriter });
-            }
-            else
-            {
-                config.TraceWriter = systemTraceWriter;
-            }
-
             config.IsSelfHost = webHostSettings.IsSelfHost;
 
             secretsRepositoryFactory = secretsRepositoryFactory ?? new DefaultSecretsRepositoryFactory();
             var secretsRepository = secretsRepositoryFactory.Create(settingsManager, webHostSettings, config);
-            _secretManager = secretManagerFactory.Create(settingsManager, config.HostConfig.LoggerFactory, secretsRepository);
+            _secretManager = secretManagerFactory.Create(settingsManager, _fallbackLoggerFactory, secretsRepository);
```

Comment:
It looks like this member is always null at this point? When is it supposed to be initialized?

Reply:
Also, rather than the disparaging "fallbackLogger" naming for this, maybe it's clearer to just name it "loggerFactory" in all these places? It's a logger factory, and in many of the places I see you using it, there isn't an alternative in scope that "fallback" distinguishes from. At least that's the way it seems to me :) The "fallback" stuff I find confusing.

Reply:
Yeah, this is busted right now (I'm fixing it).

In my local changes I've already renamed it. I wanted to be clear that this wasn't the host's logger factory -- so it wouldn't be wired up to any of the filters that the host may be using.

Reply:
BTW -- the alternative in scope is the 'LoggerProviderFactory', which is what we pass to the Host. The Host uses that to construct a LoggerFactory. We'll use that LoggerFactory provided from the Host whenever we can, but in some of these cases we need this 'other' LoggerFactory (as you've seen in v1 with the TraceWriter). I was trying to give it a conceptual name so we knew what we were using at any point -- to discriminate between the Host's and the App/WebHost/ScriptHostManager's. That's where Fallback came from -- we use it when the Host isn't there.

Reply:
I thought about it and I think I will just pass in the loggerFactory here (and call it loggerFactory). When @fabiocav gets to the DI refactor work, I think this will likely go away and we can make it clearer how this works. So, for now:
- We'll have a 'fallback/webapp/etc' LoggerFactory. But we'll just refer to it as LoggerFactory. That is to be used for logging when there's no Host LoggerFactory available.
- The LoggerProviderFactory is to be used by the Host for creating its own LoggerFactory during initialization. Only the host should use this.